### PR TITLE
matches format

### PR DESCRIPTION
**Pull request by shreymahendru**

### CODE REVIEW

File: dist/index.d.ts [hunk 1]
```diff
@@ -16,6 +16,7 @@ declare global
         base64UrlDecode(): string;
         hexEncode(): string;
         hexDecode(): string;
+        matchesFormat(format: string): string;
```
Comment (nivinjoseph) [inline dist/index.d.ts]: return type should be boolean

File: src/string-ext.ts [hunk 2]
```diff
@@ -266,4 +336,14 @@ Object.defineProperty(String.prototype, "hexDecode", {
...
         return StringExt.hexDecode(this.toString());
     }
+});
+
+Object.defineProperty(String.prototype, "matchesFormat", {
+    configurable: false,
+    enumerable: false,
+    writable: false,
+    value: function (format: string): boolean
+    {
```
Comment (nivinjoseph) [inline src/string-ext.ts:357]: if(format == null || typeof format !== "string" ||  format.trim() === "") throw new Error("format must be a valid string");

File: test/string-ext.test.ts [hunk 3]
```diff
@@ -501,4 +501,323 @@ suite("StringExt", () =>
...
+    suite.only("matchesFormat", () =>
+    {
+        suite("basic", () =>
+        {
+            test(`Given value "" and format "" when matchesFormat is called then it should return true`, () =>
+            {
+                const value = "";
+                const format = "";
+
+                assert.ok(value.matchesFormat(format));
```
Comment (nivinjoseph) [inline test/string-ext.test.ts]: Use assert.strictEquals at all times. dont use assert.ok

File: src/string-ext.ts [hunk 4]
```diff
@@ -266,4 +336,14 @@ Object.defineProperty(String.prototype, "hexDecode", {
...
+        return StringExt.matchesFormat(this.toString(), format);
```
Comment (nivinjoseph) [inline src/string-ext.ts]: trim the format before passing it in. We dont want a way to represent whitespace chars at the beginning or end.

File: src/string-ext.ts (hunk 2 above)
Comment (nivinjoseph) [reply src/string-ext.ts:357]: format also cannot contain more than one wildcard char in it. validate this (dont forget to account for escape sequence).

Comment (shreymahendru) [reply src/string-ext.ts:357]: I check multiple `*` later on when I create tokens

**Review by nivinjoseph — APPROVED** [review]

**Pull request merged**